packages/frontend-api/src/Model/Mutation/Login/LoginMutation.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Mutation\Login;
+
+use GraphQL\Error\UserError;
+use Overblog\GraphQLBundle\Definition\Argument;
+use Overblog\GraphQLBundle\Definition\Resolver\AliasedInterface;
+use Overblog\GraphQLBundle\Definition\Resolver\MutationInterface;
+use Shopsys\FrameworkBundle\Model\Customer\User\FrontendCustomerUserProvider;
+use Shopsys\FrontendApiBundle\Model\Token\TokenFacade;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+
+class LoginMutation implements MutationInterface, AliasedInterface
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Customer\User\FrontendCustomerUserProvider
+     */
+    protected $frontendUserProvider;
+
+    /**
+     * @var \Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface
+     */
+    protected $userPasswordEncoder;
+
+    /**
+     * @var \Shopsys\FrontendApiBundle\Model\Token\TokenFacade
+     */
+    protected $tokenFacade;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\FrontendCustomerUserProvider $frontendCustomerUserProvider
+     * @param \Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface $userPasswordEncoder
+     * @param \Shopsys\FrontendApiBundle\Model\Token\TokenFacade $tokenFacade
+     */
+    public function __construct(
+        FrontendCustomerUserProvider $frontendCustomerUserProvider,
+        UserPasswordEncoderInterface $userPasswordEncoder,
+        TokenFacade $tokenFacade
+    ) {
+        $this->frontendUserProvider = $frontendCustomerUserProvider;
+        $this->userPasswordEncoder = $userPasswordEncoder;
+        $this->tokenFacade = $tokenFacade;
+    }
+
+    /**
+     * @param \Overblog\GraphQLBundle\Definition\Argument $argument
+     * @return string[]
+     */
+    public function login(Argument $argument): array
+    {
+        $input = $argument['input'];
+
+        try {
+            $user = $this->frontendUserProvider->loadUserByUsername($input['email']);
+        } catch (UsernameNotFoundException $e) {
+            throw new UserError(t('Log in failed.'));
```

Comment:
I think that the API should not return translated errors and the application should take care of the translation.

Why do I think so? Because we have documentation in English and in the first phase the developer should agree with the API and if api returns the translated errors, the developer has no chance to find some information in the documentation.

Maybe we can start thinking about error codes and list of these errors in documentation.

2dd241f12ee7f74798ed079df8b046141c87aeff

Reply:
I agree, thank you :)